collector/zfs_common.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+var (
+	bsdsysctl = map[string]string{
+		"kstat.zfs.misc.abdstats.linear_cnt":             "abdstats_linear_count_total",
+		"kstat.zfs.misc.abdstats.linear_data_size":       "abdstats_linear_data_bytes",
+		"kstat.zfs.misc.abdstats.scatter_chunk_waste":    "abdstats_scatter_chunk_waste_bytes",
```

Comment:
Closing since there's no other way than to hardcode these, which can lead to discrepancies in the future as this is affecting `bsd`s for which we do not have adequate testing currently.